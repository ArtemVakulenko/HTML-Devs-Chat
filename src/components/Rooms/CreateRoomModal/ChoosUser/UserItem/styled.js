import styled from 'styled-components';

export const StUserItem = styled.div`
  display: grid;
  grid-template-columns: 2fr 8fr 2fr;
  align-items: center;
  justify-items: center;
  padding: 5px 0;
  cursor: pointer;
  &:hover{
    background: ${({ colors, theme }) => colors[theme].input};
  }
`;

export const StAvatarBlock = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  img{
  width: 38px;
  height: 38px;
  }
  span{
    margin-top: 4px;
    font-size: 12px;
    color: ${({ color, theme }) => color[theme].placeholder};
  }
`;

export const StCheck = styled.span`
  font-size: 32px;
  font-weight: 900;
  user-select: none;
`;
