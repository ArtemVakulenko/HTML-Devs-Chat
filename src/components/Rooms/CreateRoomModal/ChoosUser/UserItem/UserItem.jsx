import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { icons } from 'constants';
import { useTheme } from 'src/components/hooks';
import { StUserItem, StAvatarBlock, StCheck } from './styled';

const UserItem = ({ id, img, userName, isOnline, setChoosenUser }) => {
  const [isCheck, setCheck] = useState(false);
  const { colors, theme } = useTheme();
  const onCheck = () => {
    isCheck
      ? setChoosenUser(id, true)
      : setChoosenUser(id);
    setCheck(!isCheck);
  };
  return (
    <StUserItem colors={colors} theme={theme} onClick={onCheck}>
      <StAvatarBlock color={colors} theme={theme}>
        <img src={img || icons.defaultUserIcon} />
        <span>{isOnline ? 'Online' : 'Offline'}</span>
      </StAvatarBlock>
      <span>{userName}</span>
      <StCheck >{isCheck ? 'âœ“' : '+'}</StCheck>
    </StUserItem>
  );
};

export default UserItem;

UserItem.propTypes = {
  id: PropTypes.string.isRequired,
  img: PropTypes.string,
  userName: PropTypes.string.isRequired,
  isOnline: PropTypes.bool.isRequired,
  setChoosenUser: PropTypes.func.isRequired,
};
