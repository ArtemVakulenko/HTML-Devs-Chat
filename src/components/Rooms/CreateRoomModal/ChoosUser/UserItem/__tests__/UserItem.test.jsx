import React from 'react';
import { shallow, mount } from 'enzyme';
import UserItem from '../UserItem';

jest.mock('../../../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
        button: '',
      },
    },
    theme: 'dark',
  })),
}));

describe('UserItem', () => {
  const props = {
    id: '',
    img: '',
    userName: '',
    isOnline: false,
    setChoosenUser: jest.fn(),

  };
  it('Should match snapshot', () => {
    const component = shallow(<UserItem {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StUserItem', () => {
    const component = mount(<UserItem {...props} />);
    expect(component.find('styled__StUserItem')).toHaveLength(1);
  });
  it('Should have StAvatarBlock', () => {
    const component = mount(<UserItem {...props} />);
    expect(component.find('styled__StAvatarBlock')).toHaveLength(1);
  });
  it('Should have span', () => {
    const component = mount(<UserItem {...props} />);
    expect(component.find('span')).toHaveLength(3);
  });
  it('Should have img', () => {
    const component = mount(<UserItem {...props} />);
    expect(component.find('img')).toHaveLength(1);
  });
  it('Should have StCheck', () => {
    const component = mount(<UserItem {...props} />);
    expect(component.find('styled__StCheck')).toHaveLength(1);
  });
  it('Should have call onCheck', () => {
    const component = mount(<UserItem {...props} />);
    component.find('styled__StUserItem').simulate('click');
    expect(component.props().setChoosenUser).toHaveBeenCalled();
  });
});
