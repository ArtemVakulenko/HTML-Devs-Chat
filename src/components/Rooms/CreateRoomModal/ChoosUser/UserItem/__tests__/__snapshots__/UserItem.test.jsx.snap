// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem Should match snapshot 1`] = `"<div class=\\"styled__StUserItem-hkWVlg erntXM\\"><div color=\\"[object Object]\\" class=\\"styled__StAvatarBlock-eLyBY ctoeMe\\"><img src=\\"../assets/img/defaultUserIcon.png\\"/><span>Offline</span></div><span></span><span class=\\"styled__StCheck-gAxtRK bYeHvX\\">+</span></div>"`;
